independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/ServerRequestFilter.java

```diff
@@ -83,4 +85,14 @@
      * Whether or not the filter is a pre-matching filter
      */
     boolean preMatching() default false;
+
+    /**
+     * Normally {@link ContainerRequestFilter} classes are run by RESTEasy Reactive on the same thread as the Resource
+     * Method - this means than when a Resource Method is annotated with {@link Blocking}, the filters will also be run
+     * on a worker thread.
+     * This is meant to be set to {@code true} if a filter should be run on the event-loop even if the target Resource
+     * method is going to be run on the worker thread.
+     * For this to work, this filter must be run before any of the filters when non-blocking is not required.
+     */
+    boolean nonBlockingRequired() default false;
```

Comment:
Why do we phrase it like that?

Can we rename it to `nonBlocking` and just explain that all non-blocking filters must be first?

Reply:
I had it named `nonBlocking` at first. 
The reason I changed it is because to me it implied that the other filters that don't have the flag are blocking - which is wrong and super confusing.

WDYT?

Reply:
Ah, my bad. Here is should definitely be `nonBlocking` - the internal stuff is where I wanted the change to take place.

Reply:
Fixed